src/pages/index.js

```diff
@@ -16,79 +22,241 @@ const cardListEl = document.querySelector(".cards__list");
-  userInfo.setUserInfo(data);
-  editCardModal.close();
+profileImageEditButton.addEventListener("click", () => {
+  profilePictureModal.open();
+});
+
+addNewCardButton.addEventListener("click", () => newCardModal.open());
+
+//functions
+
+function renderCard(cardData) {
+  const card = new Card(
+    cardData,
+    cardSelector,
+    showPreviewImage,
+    handleLikeCard,
+    handleDeleteCardPopup
+  );
+  section.addItem(card.getView());
+}
+
+function handleAddCardFormSubmit({ title, url }) {
+  addCardModalSubmitButton.innerHTML = "Saving...";
+  api
+    .postCard({ name: title, link: url })
+    .then((cardData) => {
+      renderCard(cardData, cardListEl);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      addCardFormElement.reset();
+      addCardModalSubmitButton.classList.add("modal__button_disabled");
+      newCardModal.close();
+      addCardModalSubmitButton.innerHTML = "Create";
+    });
 }
 
-function handleAddCardFormSubmit(cardData) {
-  renderCard({ name: cardData.title, link: cardData.url }, cardListEl);
-  addCardFormElement.reset();
-  newCardModal.close();
+function handleDeleteCardPopup(card) {
+  deleteCardModal.open(card);
+}
+
+function handleDeleteCard(card) {
+  api
+    .deleteCard(card.getCardId())
+    .then(() => {
+      card._element.remove();
+      deleteCardModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+function handleLikeCard(card) {
+  if (card.isLiked()) {
+    api
+      .unlikeCard(card.getCardId())
+      .then(() => {
+        card._likeButton.classList.remove("card__like-button_active");
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .likeCard(card.getCardId())
+      .then(() => {
+        card._likeButton.classList.add("card__like-button_active");
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
 }
 
 function showPreviewImage(data) {
   popupWithImage.open(data);
 }
 
-profileEditButton.addEventListener("click", (e) => {
-  const userData = userInfo.getUserInfo();
-  profileTitleInput.value = userData.title;
-  profileDescriptionInput.value = userData.description;
-  editCardModal.open();
-});
+function handleProfileEditSubmit({ title, description }) {
+  profileEditSubmitButton.innerHTML = "Saving...";
+  api
+    .editUserInfo({ title, description })
+    .then(() => {
+      userInfo.setUserInfo({ name: title, about: description });
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editCardModal.close();
+      profileEditSubmitButton.innerHTML = "Save";
+    });
+}
 
-addNewCardButton.addEventListener("click", () => newCardModal.open());
+function handleChangeProfilePictureSubmit({ picture_url }) {
+  changeProfilePictureSubmitButton.innerHTML = "Saving...";
+  api
+    .editUserProfilePicture({ link: picture_url })
+    .then(() => {
+      profileImage.src = picture_url;
```

Comment:
`      profileImage.src = picture_url;
`

Class `UserInfo` should set the entire user’s data (including the **avatar**)

Reply:
When I was doing that, it was creating a problem when setuserInfo was called while editing the profile title and description in a patch method.